### PR TITLE
Added Settings Datatype

### PR DESCRIPTION
**Pull request by alexvidenov**

### CODE REVIEW

File: src/ImageSettings.hs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    , imageWidth
+    , imageHeight
+    , fieldOfView
+  ) where
+
+
+data ImageSettings = ImageSettings
+ { imageWidth :: Int
+ , imageHeight :: Int
+ , fieldOfView :: Double
```
Comment (umnikos) [inline src/ImageSettings.hs]: I'd prepend those with `get` (`getImageWidth`, `getImageHeight`, `getFieldOfView`)

File: src/Ray.hs [hunk 2]
```diff
@@ -18,3 +19,21 @@ rayMarch s end (pos,dir) -- End is the render distance (how far to march before
...
+getYComponents :: Int -> [Double]
+getYComponents height = take height [-1, -1 + 2 / fromIntegral height .. ]
+
+heightCoords :: ImageSettings -> [Double]
+heightCoords setting = getYComponents $ imageHeight setting -- Height of the image
+
+getRays :: ImageSettings -> [[Ray]]
+getRays setting = map computeRow $ heightCoords setting
+    where   z = (tan (fieldOfView setting / 2))
+            computeRow y = [ (Vec3 0 0 0, normalize (Vec3 x (-y) z) ) | x <- widthCoords setting ] -- First Ray has coordinates [-1,-(-1)].
```
Comment (umnikos) [inline src/Ray.hs]: Isn't the pull request just for the settings datatype?

File: src/ImageSettings.hs (hunk 1 above)
Comment (alexvidenov) [reply src/ImageSettings.hs]: Sure, it makes sense, since they're really just getters.

File: src/ImageSettings.hs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module ImageSettings
+  ( ImageSettings
+    , imageWidth
+    , imageHeight
+    , fieldOfView
```
Comment (umnikos) [inline src/ImageSettings.hs]: You're not exporting the constructor.

Comment (alexvidenov) [reply src/ImageSettings.hs]: True, I always forget.. Thanks for noticing!

**Review by umnikos — APPROVED** [review]
It'll need further refactoring, but it's good for now.

**Pull request merged**